saleor/tests/e2e/orders/test_cannot_fullfill_order_with_invalid_shipping_method.py

```diff
@@ -0,0 +1,174 @@
+        e2e_staff_api_client,
+        warehouse_ids=[second_warehouse_id],
+        channel_ids=[second_channel_id],
+    )
+    second_shipping_zone_id = second_shipping_zone_data["id"]
+
+    second_shipping_method_data = create_shipping_method(
+        e2e_staff_api_client, second_shipping_zone_id
+    )
+    second_shipping_method_id = second_shipping_method_data["id"]
+
+    create_shipping_method_channel_listing(
+        e2e_staff_api_client, second_shipping_method_id, second_channel_id
+    )
+
+    return (
+        first_warehouse_id,
+        first_channel_id,
+        first_shipping_method_id,
+        second_address,
+    )
+
+
+@pytest.mark.e2e
+def test_cannot_fullfill_order_with_invalid_shipping_method_core_0203(
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_product_types_and_attributes,
+    permission_manage_shipping,
+    permission_manage_orders,
+):
+    # Before
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    (
+        first_warehouse_id,
+        first_channel_id,
+        first_shipping_method_id,
+        second_address,
+    ) = prepare_shops(e2e_staff_api_client)
+
+    price = 2
+    (
+        _,
+        result_product_variant_id,
+        _,
+    ) = prepare_product(
+        e2e_staff_api_client, first_warehouse_id, first_channel_id, price
+    )
+
+    # Step 1 - Create draft order and add lines
+    draft_order_input = {
+        "channelId": first_channel_id,
+    }
+    data = draft_order_create(
+        e2e_staff_api_client,
+        draft_order_input,
+    )
+    order_id = data["order"]["id"]
+    assert order_id is not None
+
+    lines = [{"variantId": result_product_variant_id, "quantity": 1}]
+    order_lines = order_lines_create(e2e_staff_api_client, order_id, lines)
+    order_product_variant_id = order_lines["order"]["lines"][0]["variant"]["id"]
+    assert order_product_variant_id == result_product_variant_id
+
+    # Step 2 - Update order's shipping method
+    input = {
+        "shippingMethod": first_shipping_method_id,
+        "shippingAddress": first_address,
+        "billingAddress": first_address,
+    }
+    draft_order = draft_order_update(e2e_staff_api_client, order_id, input)
+    order_shipping_id = draft_order["order"]["deliveryMethod"]["id"]
+    assert order_shipping_id is not None
+
+    # Step 3 - Update order's shipping address for country PL
+    update_input = {"shippingAddress": second_address}
+    draft_update = draft_order_update(e2e_staff_api_client, order_id, update_input)
+    draft_order_shipping_id = draft_update["order"]["deliveryMethod"]["id"]
+    assert draft_order_shipping_id == order_shipping_id
+
```

Comment:
Can you add an extra check for available shipping methods? To make sure that after changing address the shipping methods list contains now the shipping method ID for the PL address.